### PR TITLE
TAB Vector (INIT): Store Phase Density Data By Columns

### PR DESCRIPTION
**Pull request by bska**
Specifically, store oil density values for every region before water density values for every region, and gas density values after water density values for every region.

This is the format expected in ECLIPSE's INIT file, and used by ResInsight's fall-back flux calculator mode for flow diagnostics.

### CODE REVIEW

**Comment by bska** [comment]
Note: This PR changes the contents of the INIT file's TAB vector.  We therefore have to update the reference solutions.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
I've also added a set of simple unit tests for this feature.

File: src/opm/output/eclipse/Tables.cpp [hunk 1]
```diff
@@ -1982,20 +1982,29 @@ namespace Opm {
...
+
+        const double default_value = -2.0e20;
+        const auto nreg = density.size();
+        const auto nph  = density[0].size;
+
+        std::vector<double> densityData(nreg * nph, default_value);
+        {
+            const auto urho = UnitSystem::measure::density;
+
+            auto* rho_o = &densityData[0*nreg + 0]; // Oil   <-> Column 0
```
Comment (atgeirr) [inline src/opm/output/eclipse/Tables.cpp]: The patch looks good, and will be merged. Just one question: why pointers and not iterators here?

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
All test errors are due to differences in TAB, as expected.

**Comment by akva2** [comment]
jenkins build this opm-tests=167 please

**Comment by atgeirr** [comment]
I would expect the INIT file changes in the opm-tests PR, but not the SMSPEC and UNSMRY changes. Do we still have something that cause those to change all the time, like a timestamp?

File: src/opm/output/eclipse/Tables.cpp (hunk 1 above)
Comment (bska) [reply src/opm/output/eclipse/Tables.cpp]: Didn't think about using iterators here, and pointers guarantee that we can use post increment without any additional cost.

Comment (bska) [reply src/opm/output/eclipse/Tables.cpp]: I can switch to using iterators if you'd like.

Comment (atgeirr) [reply src/opm/output/eclipse/Tables.cpp]: It's ok as it is. I would have merged already if not for the unclear (to me) situation with the summary output changes.

**Comment by bska** [comment]
> I would expect the INIT file changes in the opm-tests PR, but not the SMSPEC and UNSMRY changes.

Yes, those are unexpected to me too.  There is however one **possible** reason that those files change.  If the time-stepping sequence changes ever so slightly between data update runs, then the elapsed (i.e., `TIME`) parameter will also change.  Similarly, the cumulative values might deviate enough that the files as an integrated whole also create updates.

Apart from `TIME` we don't write timestamps into the summary or specificiation files.

**Comment by atgeirr** [comment]
The unexpected changes have been analysed, and the conclusion is that they stem from the new summary writer. Since the changed fields are not checked, there were no test failures and therefore no data update when that was merged.

https://github.com/OPM/opm-tests/pull/168 will be merged first, then I'll rerun this to make a more logical data update.

**Comment by akva2** [comment]
not very handy but here's the diff for formatted smspec files, left is old, right is new 

https://lh3.googleusercontent.com/-59qoF1X7tjo/Xa2n7iMt42I/AAAAAAAAEP0/ZnhgrNhNYjwq0kJprjiSW7mVWfGGra9ZQCK8BGAsYHg/s0/2019-10-21.png

**Comment by atgeirr** [comment]
> I did try to verify that there were no changes between existing and the new writer before submitting the PR

The SEQHDR field is incremented for each step in the new one. Seems better to me, although the code has a comment expressing some doubts.

**Comment by atgeirr** [comment]
jenkins build this update_data please

**Comment by akva2** [comment]
jenkins build this opm-tests=170 please

**Pull request merged**